client/rpcserver/rpcserver.go

```diff
@@ -306,6 +302,17 @@ func (s *RPCServer) Run(ctx context.Context) {
 	log.Infof("RPC server off")
 }
 
+var osExit = os.Exit
+
+func createListener(protocol string, s *RPCServer) net.Listener {
+	listener, err := tls.Listen(protocol, s.addr, s.tlsConfig)
+	if err != nil {
+		log.Errorf("can't listen on %s. rpc server quitting: %v", s.addr, err)
+		osExit(1)
```

Comment:
Using `os.Exit` doesn't allow the caller to react. Whether or not **dexc** should shut down should be up to **dexc**. Rather than use `os.Exit`, I'd suggest switching to the `dex.Connector` interface, which has an `error` return value and gives us a `WaitGroup` to use too.

Reply:
I am in middle of switching to dex.Connector interface. I think I understand how to implement use of dex.Connector but I might not understand the best way to handle a connect error.

Right now the rpc server runs from main.go via the Runner interface which does not return error so main.go is unaware if there was an rpc connect problem. That's how I landed on os.Exit.

I'm guessing all of that is understood by others more familiar with the codebase and Go.

Perhaps WaitGroup would be used in some way or I should stop using Runner interface?

Reply:
Can Runner be changed to return an error? Not sure why it's commented out. Someone on the fence?
```
// Runner is satisfied by DEX subsystems, which must start any of their
// goroutines via the Run method.
type Runner interface {
	Run(ctx context.Context) //error
}
```

Reply:
I think Runner probably could have been changed to return error. When I first started the story I assumed that it could not due to being unfamiliar with the code base.

That issue is moot now as I've converted to the dex.Connector interface.